src/modules/launcher/Plugins/Microsoft.Plugin.Calculator/CalculateEngine.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using Mages.Core;
+
+namespace Microsoft.Plugin.Calculator
+{
+    public class CalculateEngine
+    {
+        private readonly Engine _magesEngine = new Engine();
+
+        public CalculateResult Interpret(string input)
+        {
+            return Interpret(input, CultureInfo.CurrentCulture);
+        }
+
+        public CalculateResult Interpret(string input, CultureInfo cultureInfo)
+        {
+            if (input == null)
+            {
+                throw new ArgumentNullException(paramName: nameof(input));
+            }
+
+            if (!CalculateHelper.InputValid(input))
+            {
+                return default;
+            }
+
+            var result = _magesEngine.Interpret(input);
+
+            // This could happen for some incorrect queries, like pi(2)
+            if (result == null)
+            {
+                return default;
+            }
+
+            result = TransformResult(result);
+
+            if (string.IsNullOrEmpty(result?.ToString()))
+            {
+                return default;
+            }
+
+            var decimalResult = Convert.ToDecimal(result, cultureInfo);
+            var roundedResult = Math.Round(decimalResult, 10, MidpointRounding.AwayFromZero);
```

Comment:
Can you please make number of fractional digits a constant rather than hardcoding it to 10? We would ideally want this as a user setting parameter in future.

Reply:
The 10 was already hardcoded, so I have just transfered the value. Have changed it to a public const class value.